consensus/src/simplex/signing_scheme/mod.rs

```diff
@@ -0,0 +1,210 @@
+    fn can_sign(&self) -> bool;
+
+    /// Signs a vote for the given context using the supplied namespace for domain separation.
+    fn sign_vote<D: Digest>(&self, namespace: &[u8], context: VoteContext<'_, D>) -> Vote<Self>;
+
+    /// Verifies a single vote against the participant material managed by the scheme.
+    fn verify_vote<D: Digest>(
+        &self,
+        namespace: &[u8],
+        context: VoteContext<'_, D>,
+        vote: &Vote<Self>,
+    ) -> bool;
+
+    /// Batch-verifies votes and separates valid messages from the voter indices that failed
+    /// verification.
+    ///
+    /// Callers must not include duplicate votes from the same signer.
+    fn verify_votes<R, D, I>(
+        &self,
+        _rng: &mut R,
+        namespace: &[u8],
+        context: VoteContext<'_, D>,
+        votes: I,
+    ) -> VoteVerification<Self>
+    where
+        R: Rng + CryptoRng,
+        D: Digest,
+        I: IntoIterator<Item = Vote<Self>>,
+    {
+        let mut invalid = BTreeSet::new();
+
+        let verified = votes.into_iter().filter_map(|vote| {
+            if self.verify_vote(namespace, context, &vote) {
+                Some(vote)
+            } else {
+                invalid.insert(vote.signer);
+                None
+            }
+        });
+
+        VoteVerification::new(verified.collect(), invalid.into_iter().collect())
+    }
+
+    /// Aggregates a quorum of votes into a certificate, returning `None` if the quorum is not met.
+    ///
+    /// Callers must not include duplicate votes from the same signer.
+    fn assemble_certificate<I>(&self, votes: I) -> Option<Self::Certificate>
+    where
+        I: IntoIterator<Item = Vote<Self>>;
+
+    /// Verifies a certificate that was recovered or received from the network.
+    fn verify_certificate<R: Rng + CryptoRng, D: Digest>(
+        &self,
+        rng: &mut R,
+        namespace: &[u8],
+        context: VoteContext<'_, D>,
+        certificate: &Self::Certificate,
+    ) -> bool;
+
+    /// Verifies a stream of certificates, returning `false` at the first failure.
+    fn verify_certificates<'a, R, D, I>(
+        &self,
+        rng: &mut R,
+        namespace: &[u8],
+        certificates: I,
+    ) -> bool
+    where
+        R: Rng + CryptoRng,
+        D: Digest,
+        I: Iterator<Item = (VoteContext<'a, D>, &'a Self::Certificate)>,
+    {
+        for (context, certificate) in certificates {
+            if !self.verify_certificate(rng, namespace, context, certificate) {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    /// Extracts randomness seed, if provided by the scheme, derived from the certificate
+    /// for the given round.
+    fn seed(&self, round: Round, certificate: &Self::Certificate) -> Option<Self::Seed>;
+
+    /// Encoding configuration for bounded-size certificate decoding used in network payloads.
+    fn certificate_codec_config(&self) -> <Self::Certificate as Read>::Cfg;
```

Comment:
nit: can we unify `certificate_codec_config` and `certificate_codec_config_unbounded`?